public/articles/deploy-react-app-to-github-pages-guide.md

```
@@ -0,0 +1,216 @@
+
+## Setting Up Your React.js Project for GitHub Pages
+
+**Follow Step 1 Only if using Vite.**
+
+### Step 1: Configure vite.config.ts Base Path
+
+Before deploying to GitHub Pages, if you are using the default url for gh-pages https://<username>.github.io/<repository-name>/ we need to make sure that the base path is set to the repo name.
+
+Here’s how to configure your vite.config.ts:
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)